imod/typing/grid.py

```diff
@@ -279,4 +279,17 @@ def is_equal(array1: xr.DataArray, array2: xr.DataArray)->bool:
 
 @typedispatch
 def is_equal(array1: object, array2:object)->bool:
+    return False
+
+
+@typedispatch
+def is_same_domain(grid1: xu.UgridDataArray, grid2: xu.UgridDataArray)->bool:
+    return  grid1.coords.equals(grid2.coords) and grid1.ugrid.grid.equals(grid2.ugrid.grid)
```

Comment:
Is the check on grids depending on the index order? Maybe we need a call to reindex_like?

Reply:
indeed we may need a call to reindex_like, but I would leave that to the calling side.  I wouldn't expect a method like "is_same_domain" to modify the DataArray.